[DOC] Add note about int eq labels in math directive (#720)
executablebooks/jupyter-book: docs/content/math.md

```{note}

> Integer equation labels and labels which start with an integer cannot be referenced and

Reviewer: ```suggestion
Labels cannot start with an integer, or they won't be able to be referenced and
```

Reviewer: quick suggestion then I'm +1
